### PR TITLE
fix: Addressed some broken links and lacking documentation.

### LINKED ISSUE #144: Documentation Bugs: Testing and Slot Type Name
# Bug Report
* Couple of the links under the [Testing reference](https://www.litexa.com/book/testing.html#litexa-test-statements) are broken.
  * `LISTEN`
  * `database variables`
* [`user:` language reference](https://www.litexa.com/reference/#user)
  * Broken example `$name = Cat` should be `$name = "Cat"`
  * Missing example of slot values
  * Missing documentation of how to trigger `otherwise` test case (any intent in the skill)
     * Does this mean if there is no way to trigger otherwise if all intents are valid for a state?
* [Slot Builder Type naming is incorrect in the docs](https://www.litexa.com/book/state-management.html#slots) says that for a slot `$thing` it will generate a `thingType` type in my experience however (At least for slotbuilder types) it instead creates `THING` type

### PR DESCRIPTION
**Pull request by joseifert**
* Fixed broken links in the Test Statements section of the Testing chapter.
* Fixed some inaccuracies in the Language Reference.
* Added some more details to Language reference on how to test an otherwise handler.

fixes #144

### CODE REVIEW

File: packages/litexa/src/parser/litexa.pegjs [hunk 1]
```diff
@@ -1539,11 +1539,10 @@ SlotTypeStatement
   }
 
 /* litexa [otherwise]
-Defines a catch all intent that will be executed should
-its parent state receive an intent that it has no explicit
-handler for. It's always possible for the user to say
-something unexpected, so you should generally always have
-something to say to guide them back on track in these cases.
+Defines a catch all intent that will be executed should its parent state receive
```
Comment (ayrc) [inline packages/litexa/src/parser/litexa.pegjs]: catch all intent -> catch-all handler

File: packages/litexa/src/parser/litexa.pegjs [hunk 2]
```diff
@@ -1726,9 +1725,11 @@ Here is a sample IntentRequest type `$request` from running `litexa test` in a p
 ```
 ### Slot Variables
 
-The most common use of request variables is population of slot values in
-intents. If an intent contains a slot value, Litexa will automatically populate
-the $ variable defined in the handler from the skill request.
+The most common use of request variables is population of slots in intents
+(see the [with](#with) documentation for more information on slots). If a
+user invokes an intent with an utterance that has one more multiple slots in
```
Comment (ayrc) [inline packages/litexa/src/parser/litexa.pegjs]: has one more multiple slots -> has one or more slots?

File: packages/litexa/src/parser/litexa.pegjs [hunk 3]
```diff
@@ -2198,9 +2197,32 @@ For example, if a handler in the skill looks like this:
 Then the following statements behave the same way in your test:
 
 ```coffeescript
-user: "my name is Cat" # Litexa deduces $name = Cat from the utterance
-user: NameIntent with $name = Cat # append more slots by separating them with commas
-user: "my name is" with $name = Cat # this is valid, but can't happen in a real interaction
+user: "my name is Cat" # Litexa deduces $name = "Cat" from the utterance
+user: NameIntent with $name = "Cat" # append more slots by separating them with commas
```
Comment (ayrc) [inline packages/litexa/src/parser/litexa.pegjs]: oops, can you complete the example here? Maybe something like:

NameIntent with $name="Cat", $surname="Dog"

File: packages/litexa/src/parser/litexa.pegjs (hunk 1 above)
Comment (joseifert) [reply packages/litexa/src/parser/litexa.pegjs]: Polished formulation a bit - let me know if that makes sense.

File: packages/litexa/src/parser/litexa.pegjs (hunk 3 above)
Comment (joseifert) [reply packages/litexa/src/parser/litexa.pegjs]: Updated example to use 2 slots, and reformulated a bit.

File: packages/litexa/src/parser/litexa.pegjs [hunk 4]
```diff
@@ -1539,10 +1539,14 @@ SlotTypeStatement
...
-to say something unexpected, so you should generally always have something to say
-to guide them back on track in these cases.
+Defines a handler that will catch any skill intent that is not explicitly handled
+by the parent state (or the global state). `otherwise` is a great catch-all
+for anything unexpected the user might say:
+
+1. an intent that is supported elsewhere in the skill but makes no sense in the parent state
+2. an utterance that isn't supported by the skill at all, which could arrive as an
+[AMAZON.FallbackIntent](https://developer.amazon.com/blogs/alexa/post/c97f3bb7-9701-41e8-ac06-a3a44b9f1638/use-the-new-fallback-intent-to-respond-gracefully-to-unexpected-customer-requests)
+if supported by the locale the skill is running in.
```
Comment (ayrc) [inline packages/litexa/src/parser/litexa.pegjs:1549]: The word support here is a bit confusing; you still need to define the fallback intent in your skill model for it to exist, even if you aren't going to handle it explicitly in your skill (declared in an unreachable state).

Would also change the link to technical docs: https://developer.amazon.com/en-US/docs/alexa/custom-skills/standard-built-in-intents.html#fallback

File: packages/litexa/src/parser/litexa.pegjs [hunk 5]
```diff
@@ -1727,7 +1731,7 @@ Here is a sample IntentRequest type `$request` from running `litexa test` in a p
 
 The most common use of request variables is population of slots in intents
 (see the [with](#with) documentation for more information on slots). If a
-user invokes an intent with an utterance that has one more multiple slots in
+user invokes an intent with an utterance that has one more more slots in
```
Comment (ayrc) [inline packages/litexa/src/parser/litexa.pegjs:1734]: I think this still needs fixing :P

File: packages/litexa/src/parser/litexa.pegjs [hunk 6]
```diff
@@ -2174,34 +2178,36 @@ TestEndStatement
...
 ```coffeescript
-user: "my name is Cat" # Litexa deduces $name = "Cat" from the utterance
-user: NameIntent with $name = "Cat" # append more slots by separating them with commas
-user: "my name is" with $name = "Cat" # this is valid, but can't happen in a real interaction
+user: "my name is Jane Doe" # Litexa deduces $name = "Jane" and $surname = "Doe" from the utterance
+user: NameIntent with $name = "Jane", $surname = "Doe" # multiple slot values are comma-separated
 ```
 
+We recommend the second approach (using the intent name) as it makes the tests more resilient
+to changing utterances.
```
Comment (ayrc) [inline packages/litexa/src/parser/litexa.pegjs:2209]: thanks for putting this in :)

File: packages/litexa/src/parser/litexa.pegjs (hunk 4 above)
Comment (joseifert) [reply packages/litexa/src/parser/litexa.pegjs]: Take 2: Let me know what you think. Removed the `FallbackIntent` again as that might be informational overload here.

**Pull request merged**